RHDEVDOCS-3703 Pipelines 1.6.2 Release Notes (#41326)
openshift/openshift-docs: modules/op-release-notes-1-6.adoc

// https://issues.redhat.com/browse/SRVKP-1783

* The pod serving the `tkn` CLI will now be scheduled on nodes, based on the node selector and toleration limits specified in the `TektonConfig` custom resource.
// https://issues.redhat.com/browse/SRVKP-1804

[id="release-notes-1-6-2_{context}"]
== Release notes for {pipelines-title} General Availability 1.6.2

[id="fixed-issues-1-6-2_{context}"]
=== Fixed issues

> * The Operator ensures that only one instance of each type of `TektonInstallerSet` exists after an upgrade.

Author: Thanks a lot for the reference to Rolfe's presentation.

> * The Operator ensures that only one instance of each type of `TektonInstallerSet` exists after an upgrade.

Reviewer: there are now 2 installersets for clusterTasks. 1 for managing "non-versioned clustertasks" (eg. `name: buildah`) and the other for managing "versioned clustertasks" (eg. `name: buildah-1-6`, `name: buildah-1-7` etc) . The one managing versioned clustertask will not be deleted. there will be a suffix (-1-6-, -1-7- in the one which will not be deleted)

// https://issues.redhat.com/browse/SRVKP-1926

> * To decide resource recreation during an upgrade, the Operator uses the operator version instead of component version.

Reviewer 2: ```suggestion
* Before this update, all the reconcilers in the Operator used the component version to decide resource recreation during an upgrade to Red Hat OpenShift Pipelines 1.6.1 from an older version. As a result, resources were not recreated whose component versions did not change in the upgrade. With this update, the Operator uses the Operator version instead of the component version to decide resource recreation during an upgrade.
```
My bit. Feel free to improvise on this.

// https://issues.redhat.com/browse/SRVKP-1928

> * After an upgrade, the Pipelines Webhook service persists in the cluster.

Reviewer 2: ```suggestion
* Before this update, due to an update deadlock on the config maps, the Pipelines Webhook service was missing in the cluster, after an upgrade. With this update, a mechanism is added to disable webhook validation if the config maps are absent in the cluster. As a result, the Pipelines Webhook service persists in the cluster, after an upgrade.
```

// https://issues.redhat.com/browse/SRVKP-1939

> * Cron jobs for auto-prunning get recreated only if there is a relevant annotation change in the namespace.

Reviewer 2: ```suggestion
* Before this update, cron jobs for auto-pruning got recreated, after any configuration change to the namespace. With this update, cron jobs for auto-pruning get recreated only if there is a relevant annotation change in the namespace.
```

// https://issues.redhat.com/browse/SRVKP-1826

> * Upstream version of Tekton Pipelines has been revised to `v0.28.3`, which has the following fixes:

Reviewer 2: ```suggestion
* The upstream version of Tekton Pipelines has been revised to `v0.28.3`, which has the following fixes:
```

> ** Fix pipeline/task to Run label/annotation propagation.

Reviewer 2: ```suggestion
** Fix pipeline or task to Run label or annotation propagation.
```

** For implicit params:
*** Do not apply the `PipelineSpec` params to TaskRefs.
*** Disable implicit param behavior for `Pipeline` objects.

> // https://github.com/tektoncd/pipeline/releases/tag/v0.28.3

Reviewer 2: Observation: I see this line on the GH page: [#4521](https://github.com/tektoncd/pipeline/pull/4521) Update Dockerfiles using golang images to Go 1.16.13

I believe that you have not mentioned it in the RN as it is not relevant. Please recheck and confirm. Thanks!

Author: Yes, it's not relevant based on SME Input.

> * THe upstream version of Tekton Pipelines has been revised to `v0.28.3`, which has the following fixes:

Reviewer 2: ```suggestion
* The upstream version of Tekton Pipelines has been revised to `v0.28.3`, which has the following fixes:
```

> ** Fix PipelineRun/TaskRun objects to label/annotation propagation.

Reviewer 2: ```suggestion
** Fix PipelineRun or TaskRun objects to label or annotation propagation.
```
Question: Is the use of "/" intentional here? Can this be replaced with "or"?

Author: Yes, / can be substituted by "or".

> *** Disable implicit param behavior for `Pipeline` objects.

Reviewer 2: ```suggestion
*** Disable implicit param behavior for the `Pipeline` objects.
```

// https://issues.redhat.com/browse/SRVKP-1783

* The pod serving the `tkn` CLI will now be scheduled on nodes, based on the node selector and toleration limits specified in the `TektonConfig` custom resource.
// https://issues.redhat.com/browse/SRVKP-1804

[id="release-notes-1-6-2_{context}"]
== Release notes for {pipelines-title} General Availability 1.6.2

[id="known-issues-1-6-2_{context}"]
=== Known issues

> * When you create a new project, the creation of service account is delayed, and removal of existing cluster tasks and pipeline templates takes more than 10 minutes.

Reviewer 3: ```suggestion
* When you create a new project, the creation of the `pipeline` service account is delayed, and removal of existing cluster tasks and pipeline templates takes more than 10 minutes.
```

Author: Resolved.
